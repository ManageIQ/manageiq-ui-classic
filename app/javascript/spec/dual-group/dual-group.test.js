import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import DualGroup from '../../components/dual-group';

jest.mock('@@ddf', () => ({
  useFormApi: () => ({
    renderForm: ([{ name }]) => `hello ${name}`,
  }),
}));


describe('Dual group component', () => {
  const initialProps = {
    name: 'dualgroup',
    fields: [
      { name: 'John' },
      { name: 'Arthur' },
    ],
  };

  const origConsole = [];

  beforeEach(() => {
    origConsole.error = window.console.error;
    window.console.error = jest.fn();
  });

  afterEach(() => {
    window.console.error = origConsole.error;
  });

  it('should render', () => {
    const wrapper = mount(<DualGroup {...initialProps} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should throw error', () => {
    try {
      mount(<DualGroup {...initialProps} fields={[...initialProps.fields, { name: 'Dutch' }, { name: 'Micah' }, { name: 'Hosea' }]} />);
    } catch (error) {
      expect(error.message).toEqual('Length of fields (DualGroup component: dualgroup) has to be a divisor of 12: 1,2,3,4,6,12. You have: 5');
    }
  });
});
