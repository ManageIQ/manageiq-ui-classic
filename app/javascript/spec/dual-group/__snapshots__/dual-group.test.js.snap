// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dual group component should render 1`] = `
<Dualgroup
  fields={
    Array [
      Object {
        "name": "John",
      },
      Object {
        "name": "Arthur",
      },
    ]
  }
  name="dualgroup"
>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <div
      className="row"
    >
      <Col
        bsClass="col"
        componentClass="div"
        key="John"
        md={6}
      >
        <div
          className="col-md-6"
        >
          hello John
        </div>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        key="Arthur"
        md={6}
      >
        <div
          className="col-md-6"
        >
          hello Arthur
        </div>
      </Col>
    </div>
  </Row>
  <hr
    style={
      Object {
        "margin": 0,
      }
    }
  />
</Dualgroup>
`;
